templates/server/src/main/resources/archetype-resources/core/src/main/java/__packageInPathFormat__/general/service/impl/config/BaseWebSecurityConfig.java

```diff
@@ -74,23 +77,23 @@ public void configure(HttpSecurity http) throws Exception {
         .userDetailsService(this.userDetailsService)
         // define all urls that are not to be secured
         .authorizeRequests().antMatchers(unsecuredResources).permitAll().anyRequest().authenticated().and()
-
-        // activate crsf check for a selection of urls (but not for login & logout)
-        .csrf().requireCsrfProtectionMatcher(new CsrfRequestMatcher()).and()
-
         // configure parameters for simple form login (and logout)
         .formLogin().successHandler(new SimpleUrlAuthenticationSuccessHandler()).defaultSuccessUrl("/")
         .failureUrl("/login.html?error").loginProcessingUrl("/j_spring_security_login").usernameParameter("username")
         .passwordParameter("password").and()
         // logout via POST is possible
         .logout().logoutSuccessUrl("/login.html").and()
-
         // register login and logout filter that handles rest logins
         .addFilterAfter(getSimpleRestAuthenticationFilter(), BasicAuthenticationFilter.class)
         .addFilterAfter(getSimpleRestLogoutFilter(), LogoutFilter.class);
 
+	//disable CSRF protection by default, use csrf starter to override.
+	http = http.csrf().disable();
+	//load starters as pluggins.
+    http = this.webSecurityConfigurer.configure(http);
```

Comment:
please move to the beginning of this method so the custom config code comes after and allows overriding by usere.

Reply:
also fix indendation please